File: Ad Hoc/4307번 - 개미/4307.cpp
#include <iostream>

using namespace std;

int l, n;
int ant[100001] = { 0, };

void input() {
    cin >> l >> n;
    for(int i = 0; i < n; i++) {
        cin >> ant[i];
    }
}

void solve() {
    int ans_min = 0, ans_max = 0;
    
    while(n--) {
        int sht = min(ant[n], l - ant[n]), lon = max(ant[n], l - ant[n]);
        ans_min = max(ans_min, sht);
        ans_max = max(ans_max, lon);
    }
    
    cout << ans_min << ' ' << ans_max << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    int tc = 0;
    cin >> tc;
    while(tc--) {
        input();
        solve();
    }
    
    return 0;
}

File: Ad Hoc/4307번 - 개미/4307.py
import sys

input = sys.stdin.readline

def solve(l, n, ant):
    ans_min, ans_max = 0, 0
    
    while n:
        n -= 1
        sht, lon = min(ant[n], l - ant[n]), max(ant[n], l - ant[n])
        ans_min = max(ans_min, sht)
        ans_max = max(ans_max, lon)
        
    print(ans_min, ans_max)


if __name__ == '__main__':
    tc = int(input())
    while tc:
        l, n = map(int, input().split())
        ant = [int(input()) for _ in range(n)]
        solve(l, n, ant)
        tc -= 1